File: Labwork 4/C#/Part2.cs
using System;

class Program
{
    static void Main()
    {
        int x = 10;
        Console.WriteLine($"{x} ({x.GetType()})"); 


        dynamic y = 10;
        Console.WriteLine($"{y} ({y.GetType()})");  

        y = "ten";
        Console.WriteLine($"{y} ({y.GetType()})");  

        y = true;
        Console.WriteLine($"{y} ({y.GetType()})");  
    }
}


// • What happens in each case?
//      Using 'int x' enforces static typing — reassigning to a different type fails at compile time.
//      Using 'dynamic y' allows changing types at runtime, similar to Python.
//
// • If the compiler/interpreter allows it, print type(x) or equivalent.
//      For 'dynamic', y.GetType() returns the runtime type.
//

File: Labwork 4/typescript/Part2.ts

let x: number = 10;
console.log(x, typeof x); // 10 number


let y: any = 10;
console.log(y, typeof y); 

y = "ten";
console.log(y, typeof y); 

y = true;
console.log(y, typeof y); 



// • What happens in each case?
//     TypeScript enforces static typing. Reassigning 'x' to a different type causes a compile-time error.
//      If declared as 'any', 'y' can change type freely at runtime (like JavaScript).

// • If the compiler/interpreter allows it, print type(x) or equivalent.
//      typeof y prints the runtime JavaScript type (number, string, boolean).
